libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationTest.java

```diff
@@ -28,199 +28,251 @@
     MapboxNavigationOptions options = MapboxNavigationOptions.builder().defaultMilestonesEnabled(false).build();
-    MapboxNavigation navigationWithOptions = new MapboxNavigation(mock(Context.class),
-      ACCESS_TOKEN, options, mock(NavigationTelemetry.class), mock(LocationEngine.class));
+    MapboxNavigation navigationWithOptions = buildMapboxNavigationWithOptions(options);
+
     assertEquals(0, navigationWithOptions.getMilestones().size());
   }
 
   @Test
-  public void defaultEngines_didGetInitialized() throws Exception {
-    assertNotNull(navigation.getSnapEngine());
+  public void defaultEngines_offRouteEngineDidGetInitialized() throws Exception {
+    MapboxNavigation navigation = buildMapboxNavigation();
+
     assertNotNull(navigation.getOffRouteEngine());
   }
 
+  @Test
+  public void defaultEngines_snapEngineDidGetInitialized() throws Exception {
+    MapboxNavigation navigation = buildMapboxNavigation();
+
+    assertNotNull(navigation.getSnapEngine());
+  }
+
   @Test
   public void offRouteEngine_doesNotGetInitializedWithFalseOption() throws Exception {
     MapboxNavigationOptions options = MapboxNavigationOptions.builder()
       .enableOffRouteDetection(false)
       .build();
-    navigation = new MapboxNavigation(mock(Context.class), ACCESS_TOKEN, options, mock(NavigationTelemetry.class),
-      mock(LocationEngine.class));
-    assertNull(navigation.getOffRouteEngine());
+    MapboxNavigation navigationWithOptions = buildMapboxNavigationWithOptions(options);
+
+    assertNull(navigationWithOptions.getOffRouteEngine());
   }
 
   @Test
   public void snapToRouteEngine_doesNotGetInitializedWithFalseOption() throws Exception {
     MapboxNavigationOptions options = MapboxNavigationOptions.builder()
       .snapToRoute(false)
       .build();
-    navigation = new MapboxNavigation(mock(Context.class), ACCESS_TOKEN, options, mock(NavigationTelemetry.class),
-      mock(LocationEngine.class));
-    assertNull(navigation.getSnapEngine());
+    MapboxNavigation navigationWithOptions = buildMapboxNavigationWithOptions(options);
+
+    assertNull(navigationWithOptions.getSnapEngine());
   }
 
   @Test
   public void fasterRouteEngine_doesNotGetInitializedWithFalseOption() throws Exception {
     MapboxNavigationOptions options = MapboxNavigationOptions.builder()
       .enableFasterRouteDetection(false)
       .build();
-    navigation = new MapboxNavigation(mock(Context.class), ACCESS_TOKEN, options, mock(NavigationTelemetry.class),
-      mock(LocationEngine.class));
-    assertNull(navigation.getFasterRouteEngine());
+    MapboxNavigation navigationWithOptions = buildMapboxNavigationWithOptions(options);
+
+    assertNull(navigationWithOptions.getFasterRouteEngine());
   }
 
   @Test
   public void addMilestone_milestoneDidGetAdded() throws Exception {
+    MapboxNavigation navigation = buildMapboxNavigation();
+
     Milestone milestone = new StepMilestone.Builder().build();
```

Comment:
IMO These two lines of code belongs to the `Arrange` part. 

This applies to other tests included in `MapboxNavigationTest`.